Java-wasm-allocator/src/main/org/m4d/adp/allocator/WasmAllocationManager.java

```diff
@@ -0,0 +1,59 @@
+package org.m4d.adp.allocator;
+
+
+import org.apache.arrow.memory.AllocationManager;
+import org.apache.arrow.memory.BufferAllocator;
+
+
+/**
+ * Allocation manager that uses allocation and deallocation functions from a WASM module.
+ */
+
+public final class WasmAllocationManager extends AllocationManager {
+
+    public static final WasmAllocationFactory FACTORY = new WasmAllocationFactory();
```

Comment:
I don't think it should be here. It should probably be creates explicitly by the user so it can actually be closed.